src/text-editor.js

```diff
@@ -400,6 +401,16 @@ class TextEditor {
           }
           break
 
+        case 'readOnly':
+          if (value !== this.readOnly) {
+            this.readOnly = value
+            if (this.component != null) {
+              this.component.scheduleUpdate()
+            }
+            this.buffer.emitModifiedStatusChanged(this.isModified())
```

Comment:
Should not emit "modified state changed" event when it's really not changed.
If some feature need hook for this, fire distinct event like `did-set-read-only` and do manual work in this.

Reply:
I noticed this part is reverted in ca53cf9. sorry.